chain-impl-mockchain/src/ledger/evm.rs

```diff
@@ -52,12 +162,24 @@ impl EvmState for super::Ledger {
 }
 
 impl super::Ledger {
-    pub fn run_transaction(
-        &mut self,
-        contract: EvmTransaction,
-        config: Config,
-    ) -> Result<(), Error> {
-        let config = config.into();
+    pub fn apply_map_accounts<'a>(
+        mut self,
+        tx: &EvmMapping,
+        auth_data: &TransactionBindingAuthData<'a>,
+        sig: SingleAccountBindingSignature,
+    ) -> Result<Self, Error> {
+        if sig.verify_slice(&tx.account_id().clone().into(), auth_data) != Verification::Success {
+            return Err(Error::EvmMappingSignatureFailed);
+        }
+
+        self.evm
+            .address_mapping
+            .map_accounts(tx.account_id().clone(), *tx.evm_address())?;
+        Ok(self)
+    }
+
+    pub fn run_transaction(mut self, contract: EvmTransaction) -> Result<Self, Error> {
```

Comment:
should this at the end remove funds from the `JorAccount` if the balance is shared?

Reply:
All accounts mutation is appeared inside `modify_account` directly